features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/configureroom/ConfigureRoomPresenter.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.features.createroom.impl.configureroom
+
+import android.net.Uri
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import io.element.android.libraries.architecture.Presenter
+import kotlinx.collections.immutable.toImmutableList
+
+class ConfigureRoomPresenter @AssistedInject constructor(
+    @Assisted val args: ConfigureRoomPresenterArgs,
+) : Presenter<ConfigureRoomState> {
+
+    @AssistedFactory
+    interface Factory {
+        fun create(args: ConfigureRoomPresenterArgs): ConfigureRoomPresenter
+    }
+
+    @Composable
+    override fun present(): ConfigureRoomState {
+        var roomName by rememberSaveable { mutableStateOf("") }
+        var topic by rememberSaveable { mutableStateOf("") }
+        var avatarUri by rememberSaveable { mutableStateOf<Uri?>(null) }
+        var privacy by rememberSaveable { mutableStateOf<RoomPrivacy?>(null) }
+        val isCreateButtonEnabled by rememberSaveable(roomName, privacy) {
```

Comment:
I think it's better to use `derivedStateOf` in the ConfigureRoomView, just by adding:

```kotlin
    val isCreateButtonEnabled by remember {
        derivedStateOf {
            state.roomName.isNotEmpty() && state.privacy != null
        }
    }
```

and remove `ConfigureRoomState::isCreateButtonEnabled`

But note that the value cannot be tested anymore in `ConfigureRoomPresenterTests` so I am not sure what's best... @ganfra @jmartinesp WDYT?

Reply:
done https://github.com/vector-im/element-x-android/pull/308/commits/75939c1118e7db7691b0b979dc6269f3578b2ef2